cwf/cloud/go/services/cwf/obsidian/models/conversion.go

```diff
@@ -228,3 +230,66 @@ func (m *LiUes) GetFromNetwork(network configurator.Network) interface{} {
 func (m *LiUes) ValidateModel() error {
 	return m.Validate(strfmt.Default)
 }
+
+func (m *CwfHaPair) ToEntity() configurator.NetworkEntity {
+	return configurator.NetworkEntity{
+		Type:   cwf.CwfHAPairType,
+		Key:    m.HaPairID,
+		Config: m.Config,
+		Associations: []storage.TypeAndKey{
+			{
+				Type: cwf.CwfGatewayType,
+				Key:  m.GatewayID1,
+			},
+			{
+				Type: cwf.CwfGatewayType,
+				Key:  m.GatewayID2,
+			},
+		},
+	}
+}
+
+func (m *CwfHaPair) FromBackendModels(ent configurator.NetworkEntity) (*CwfHaPair, error) {
+	gatewayIDs := []string{}
+	for _, assoc := range ent.Associations {
+		if assoc.Type != cwf.CwfGatewayType {
+			continue
+		}
+		gatewayIDs = append(gatewayIDs, assoc.Key)
+	}
+	if len(gatewayIDs) != 2 {
+		return &CwfHaPair{}, fmt.Errorf("could not convert entity to CwfHaPair; could not parse gateway pair IDs")
+	}
+	if ent.Config == nil {
+		return &CwfHaPair{}, fmt.Errorf("could not convert entity to CwfHaPair; config was nil")
+	}
+	cfg, ok := ent.Config.(*CwfHaPairConfigs)
+	if !ok {
+		return &CwfHaPair{}, fmt.Errorf("could not convert entity to CwfHaPair; config conversion failed")
+	}
+	return &CwfHaPair{
+		HaPairID:   ent.Key,
+		GatewayID1: gatewayIDs[0],
+		GatewayID2: gatewayIDs[1],
+		Config:     cfg,
+	}, nil
```

Comment:
you should set these in-place on the receiver, otherwise this should just be a flat function. You can return the receiver for ergonomics but you'll be incompatible with some interfaces like `PartialEntityModel`